### PR TITLE
Fixed issue #16937: Easy way to launch CDBException

### PR DESCRIPTION
**Pull request by gabrieljenik**
Porting #16481 to LS4.
This is the second PR for this ticket.
The first one (https://github.com/LimeSurvey/LimeSurvey/pull/1708) has usefull comments but had to be closed without merge as it was harde to do it (got old and merging was dificult).

### CODE REVIEW

File: application/helpers/expressions/em_manager_helper.php [hunk 1]
```diff
@@ -9800,6 +9752,75 @@ public static function htmlSpecialCharsUserValue($string)
...
+                        break;
+                }
+                $_SESSION[$LEM->sessid][$knownVar['sgqa']] = $value;
+                $LEM->updatedValues[$knownVar['sgqa']] = array(
+                    'type' => $knownVar['type'],
+                    'value' => $value,
+                );
+                unset($_SESSION[$LEM->sessid]['startingValues'][$k]);
+            }
+            $LEM->_UpdateValuesInDatabase();
```
Comment (Shnoulle) [inline application/helpers/expressions/em_manager_helper.php:9821]: Mybe unset `$_SESSION[$LEM->sessid]['startingValues']` at end ?

Comment (gabrieljenik) [reply application/helpers/expressions/em_manager_helper.php:9821]: This is the port of the PR already tested and honestly adding that line seems to terminative. 
Wouldn't be that terminative :)

I would leave it like this.

**Comment by cdorin93** [comment]
@gabrieljenik , can you please make the PR for master?

**Comment by gabrieljenik** [comment]
This is the error. Let me review it.
```
1) ls\tests\PreviewGroupAndQuestionTest::testPreviewGroupPrefill
Prefilling url broken when preview group
Failed asserting that false is true.
```

**Comment by cdorin93** [comment]
removed the tested-ok status so that Gabriel can check that error :)

**Comment by Shnoulle** [comment]
Yes : PreviewGroupAndQuestionTest use prefilling value for Expressionmanager.

Since it was only when preview is set (and no update database can/must happen) : maybe just check if we are in https://github.com/LimeSurvey/LimeSurvey/blob/279c606f54d6865f3ad1f6081a4ec91befff34de/application/helpers/SurveyRuntimeHelper.php#L52 mode ?

**Comment by gabrieljenik** [comment]
> st check if we are in

Will review. Thanks!

**Comment by gabrieljenik** [comment]
SaveStartingValues was moved from StartSurvey to InitMove. InitMove is not being called from the preview mode. So now, we call SaveStartingValues from the preview.

**Comment by olleharstedt** [comment]
Someone will test this?

**Comment by gabrieljenik** [comment]
Seems Travis was green for a few but red for others.
From the logs I see a wget error.
Will just send a new commit to restart the tests

**Comment by olleharstedt** [comment]
Yes, sometimes Travis fails internally, not because of bugs in the code.

**Comment by olleharstedt** [comment]
Bug fix to develop? Or should be to master?

**Comment by gabrieljenik** [comment]
Should be master.

**Comment by olleharstedt** [comment]
OK, will close this one. Sorry.

**Comment by gabrieljenik** [comment]
New PR: https://github.com/LimeSurvey/LimeSurvey/pull/1754

**Comment by olleharstedt** [comment]
Huzzah!

**Pull request closed without merging**